core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -413,8 +413,6 @@ class KafkaApis(val requestChannel: RequestChannel,
         val topicPartition = new TopicPartition(topicName, partition.index())
         if (topicName.isEmpty)
```

Comment:
This is another subtle existing issue. If a produce request before v13 has an empty topic name, we used to return UNKNOWN_TOPIC_OR_PARTITION. Now, we return UNKNOWN_TOPIC_ID. It's probably better to change the condition to `if (topicName.isEmpty && topic.topicId().equals(Uuid.ZERO_UUID))`.

Reply:
>  It's probably better to change the condition to if (topicName.isEmpty && topic.topicId().equals(Uuid.ZERO_UUID)).

The default value of `topic.topicId()` is `Uuid.ZERO_UUID, and hence `if (topicName.isEmpty && topic.topicId().equals(Uuid.ZERO_UUID))` is always true if request is before v13. Maybe we can try another condition: `if (topicName.isEmpty && version > 12`?

Reply:
Sorry, I meant `if (topicName.isEmpty && !topic.topicId().equals(Uuid.ZERO_UUID))`.